examples/api/lib/rendering/growth_direction/growth_direction.0.dart

```diff
@@ -0,0 +1,244 @@
+        axis = Axis.vertical;
+      case AxisDirection.down:
+        reverse = false;
+        axis = Axis.vertical;
+      case AxisDirection.left:
+        reverse = true;
+        axis = Axis.horizontal;
+      case AxisDirection.right:
+        reverse = false;
+        axis = Axis.horizontal;
+      case null:
+    }
+    setState((){
+      // Respond to change in axis direction.
+    });
+  }
+
+  Widget _getLeading(SliverConstraints constraints, bool isForward) {
+    return Container(
+      color: isForward ? Colors.orange[300] : Colors.green[400],
+      child: Padding(
+        padding: const EdgeInsets.all(8.0),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: <Widget>[
+            Text(constraints.axis.toString()),
+            spacer,
+            Text(constraints.axisDirection.toString()),
+            spacer,
+            Text(constraints.growthDirection.toString()),
+            spacer,
+            _getArrows(constraints),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget _getRadioRow() {
+    return DefaultTextStyle(
+      style: const TextStyle(fontWeight: FontWeight.bold, color: Colors.white),
+      child: RadioTheme(
+        data: RadioThemeData(
+          fillColor: MaterialStateProperty.all<Color>(Colors.white),
+        ),
+        child: Padding(
+          padding: const EdgeInsets.all(8.0),
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.spaceAround,
+            children: <Widget>[
+              Radio<AxisDirection>(
+                value: AxisDirection.up,
+                groupValue: _getAxisDirection(),
+                onChanged: _onAxisDirectionChanged,
+              ),
+              const Text('up'),
+              spacer,
+              Radio<AxisDirection>(
+                value: AxisDirection.down,
+                groupValue: _getAxisDirection(),
+                onChanged: _onAxisDirectionChanged,
+              ),
+              const Text('down'),
+              spacer,
+              Radio<AxisDirection>(
+                value: AxisDirection.left,
+                groupValue: _getAxisDirection(),
+                onChanged: _onAxisDirectionChanged,
+              ),
+              const Text('left'),
+              spacer,
+              Radio<AxisDirection>(
+                value: AxisDirection.right,
+                groupValue: _getAxisDirection(),
+                onChanged: _onAxisDirectionChanged,
+              ),
+              const Text('right'),
+              spacer,
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _getList({ required bool isForward }) {
+    // The SliverLayoutBuilder is not necessary, and is here to allow us to
+    // the SliverContraints & directional information that is provided to the
+    // SliverList when laying out. Since it is the top level sliver here, it's
+    // key will be center if it is the forward list.
```

Comment:
Hm, what does this mean? The LayoutBuilder is not assigned a key?

Reply:
Leftover from earlier version, cleaned up, thanks!